UnitystationLauncher/Models/ServerManager.cs

```diff
@@ -68,6 +68,12 @@ private async Task<IReadOnlyList<ServerViewModel>> GetServerList()
                 }
                 else
                 {
+                    if (!HasTrustedUrlSource(s))
```

Comment:
This is the wrong location for the check, since this branch is only taken when the server is newly added to the server list.
You could still create the server with a trusted URL and change it to a malicious one.

Reply:
done